stake-pool/program/src/state.rs

```diff
@@ -0,0 +1,102 @@
+//! State transition types
+
+use crate::error::Error;
+use crate::instruction::{unpack, Fee};
+use solana_sdk::{entrypoint::ProgramResult, program_error::ProgramError, pubkey::Pubkey};
+use std::mem::size_of;
+
+/// Initialized program details.
+#[repr(C)]
+#[derive(Clone, Copy, Debug, Default, PartialEq)]
+pub struct StakePool {
+    /// Owner authority
+    /// allows for updating the staking authority
+    pub owner: Pubkey,
+    /// Deposit authority nonce
+    /// for `create_program_address(&[state::StakePool account, "deposit"])`
+    pub deposit_nonce: u8,
+    /// Withdrawal authority nonce
+    /// for `create_program_address(&[state::StakePool account, "withdrawal"])`
+    pub withdraw_nonce: u8,
```

Comment:
Why are there two different authorities?  It seems like this could all be done with just one

Reply:
I need a way to track which stakes have been deposited and which have not.